extensions/Worker.Extensions.Http.AspNetCore/src/HostBuilderExtensions.cs

```diff
@@ -25,15 +38,85 @@ public static IHostBuilder ConfigureAspNetCoreIntegration(this IHostBuilder buil
             builder.ConfigureWebHostDefaults(webBuilder =>
             {
                 webBuilder.UseUrls(HttpUriProvider.HttpUriString);
+
                 webBuilder.Configure(b =>
                 {
+                    b.UseRouting();
                     b.UseMiddleware<WorkerRequestServicesMiddleware>();
+
                     // TODO: provide a way for customers to configure their middleware pipeline here
+
+                    b.UseEndpoints(endpoints =>
+                    {
+                        string scriptRoot = Environment.GetEnvironmentVariable(FunctionsApplicationDirectoryKey) ??
+                           throw new InvalidOperationException("Cannot determine script root directory.");
+
+                        var metadataProvider = b.ApplicationServices.GetRequiredService<IFunctionMetadataProvider>();
+                        var metadata = metadataProvider.GetFunctionMetadataAsync(scriptRoot).GetAwaiter().GetResult();
+
+                        string routePrefix = GetRoutePrefixFromHostJson(scriptRoot) ?? DefaultRoutePrefix;
+
+                        foreach (var functionMetadata in metadata)
+                        {
+                            MapHttpFunction(endpoints, functionMetadata, routePrefix);
+                        }
+                    });
+
                     b.UseMiddleware<InvokeFunctionMiddleware>();
                 });
             });
 
             return builder;
         }
+
+        internal static void MapHttpFunction(IEndpointRouteBuilder endpoints, IFunctionMetadata functionMetadata, string routePrefix)
+        {
+            if (functionMetadata.RawBindings is null)
+            {
+                return;
+            }
+
+            var functionName = functionMetadata.Name ?? string.Empty;
+
+            foreach (var binding in functionMetadata.RawBindings)
+            {
+                var functionBinding = JsonSerializer.Deserialize<FunctionHttpBinding>(binding, _jsonSerializerOptions);
```

Comment:
Note that I will follow up with a better way to manage this. Today we return a json string to send back to the host... but that stinks for us as we now have to deserialize it.

We need to fix this up in the core package but we don't want to block next preview on it.